webhooky/tests/auth.rs

```diff
@@ -0,0 +1,741 @@
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/github/verify", port))
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::UNAUTHORIZED);
+}
+
+#[tokio::test]
+async fn test_missing_signature_hmac_audit_passes() {
+    let (port, _server) = make_server();
+
+    let test_body = include_str!("../tests/github_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/github/audit", port))
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::ACCEPTED);
+}
+
+/// Test GitHub signatures
+
+#[tokio::test]
+async fn test_github_hmac_passes() {
+    let (port, _server) = make_server();
+
+    let test_signature = "sha256=8c7ac6b6a1ca30229207b4406d50b5c034d90f56009835bc7f32a16b2044d29d";
+    let test_body = include_str!("../tests/github_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/github/verify", port))
+        .header("X-Hub-Signature-256", test_signature)
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::ACCEPTED);
+}
+
+#[tokio::test]
+async fn test_github_hmac_fails() {
+    let (port, _server) = make_server();
+
+    let test_signature = "sha256=8c7ac6b6a1ca30229207b4406d50b5c034d90f56009835bc7f32a16b2044d29c";
+    let test_body = include_str!("../tests/github_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/github/verify", port))
+        .header("X-Hub-Signature-256", test_signature)
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::UNAUTHORIZED);
+}
+
+#[tokio::test]
+async fn test_github_hmac_audit_passes_with_invalid_signature() {
+    let (port, _server) = make_server();
+
+    let test_signature = "sha256=8c7ac6b6a1ca30229207b4406d50b5c034d90f56009835bc7f32a16b2044d29c";
+    let test_body = include_str!("../tests/github_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/github/audit", port))
+        .header("X-Hub-Signature-256", test_signature)
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::ACCEPTED);
+}
+
+/// Test Checkr signatures
+
+#[ignore]
```

Comment:
As you mentioned below, the `Checkr` key unlike the others comes from a db for historical reasons. I've taken the suggestion though to support both an env var and a db value so that these tests could be actually implemented.